tests/c/CMakeLists.txt

```diff
@@ -1,14 +1,16 @@
-cmake_minimum_required(VERSION 3.20)
 
-if(USE_VCPKG AND NOT ANDROID)
+if(BUILD_TESTS)
+cmake_minimum_required(VERSION 3.20)
```

Comment:
This was probably a good idea fair play. The high version for Cmake was to resolve some issue on Mac for GTest as far as I can remember. I agree it's a bit high for some platforms. Maybe we can set it lower except on Mac to overcome that. I think Steven said that 3.16 was the common highest. I'll see if I can recall the issue and work on dropping it down.